homeassistant/components/hunterdouglas_powerview/sensor.py

```diff
@@ -111,15 +95,60 @@ async def async_setup_entry(
         name_before_refresh = shade.name
         room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
         room_name = pv_entry.room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
-        for cls, attr in SENSOR_TYPES.items():
-            if attr in shade.raw_data:
+
+        for description in SENSORS:
+            if description.create_sensor_fn(shade):
                 entities.append(
-                    cls(
+                    PowerViewSensor(
                         pv_entry.coordinator,
                         pv_entry.device_info,
                         room_name,
                         shade,
                         name_before_refresh,
+                        description,
                     )
                 )
+
     async_add_entities(entities)
+
+
+class PowerViewSensor(ShadeEntity, SensorEntity):
+    """Representation of an shade sensor."""
+
```

Comment:
```suggestion
    entity_description: PowerviewSensorDescription
```

black will complain about this suggestion

Reply:
black isnt complaining.... what would you expect this to look like ?

Reply:
It might want another empty line

Reply:
> was there a reason for adding this here and not on the select and button equivalent?

I just missed it on the other ones.

> we were already calling PowerviewSensorDescription on line 126 when setting description (same method used in button and select)

Generally we want to override type hints above instead of in `__init__`. While both work, its a hint to the dev that we are overriding something vs adding missing typing.

Reply:
thanks for the info - appreciate the explanation of why its better that way
I've made the requested changes and removed the breaking change as requested below